src/utils/shuffle.ts

```diff
@@ -0,0 +1,53 @@
+import { v5 as uuidv5 } from 'uuid';
+
+export function getRandomValue(seed: string, max: number): number {
+  if (max === 0) {
+    return 0;
+  }
+
+  let hash = 0;
+  // eslint-disable-next-line no-plusplus
+  for (let i = 0; i < seed.length; i++) {
+    hash = hash * 31 + seed.charCodeAt(i);
+    // eslint-disable-next-line no-bitwise
+    hash |= 0; // Convert to 32bit integer
+  }
+  // max is never attained
+  return Math.abs(hash) % max;
+}
```

Comment:
Can you give more details about how the random number generator works ? which algorithm it is uses etc.

Reply:
Here's the breakdown. Should I add it as comments?

**Hash calculation**: It initializes a variable hash to 0. Then, it iterates over each character in the seed string, and for each character, it updates the hash using a simple hashing algorithm. It multiplies the current hash value by 31 and adds the Unicode value of the character at the current position in the seed string. Finally, it bitwise ORs the hash with 0 to ensure it's treated as a 32-bit integer.

**Pseudo-random number generation**: After hashing the seed, it takes the absolute value of the resulting hash value (to ensure it's non-negative) and calculates the remainder when dividing it by max using the modulo operator %. This operation ensures that the result is within the range [0, max).